src/app/button-page/button-page.component.html

```diff
@@ -1,14 +1,14 @@
 <div class="board" *ngIf="board; else noBoard">
   <div class="boardRow"
-      [class.scanHighlight]="currentScanHighlightRow(row)"
-      [class.scanSelect]="currentScanSelectRow(row)"
+      [class.scanHighlight]="scanningModel && scanningModel.currentHighlight === row"
```

Comment:
It depends what you want to do with the scanningModel. At the moment the code is here because this has access to the scanningModel, whereas the row does not. I did play around with moving the scanningModel into the row provider and passing it down to the rows and the buttons, but it seemed a lot messier than doing it here.

Reply:
ANother option could be to call the row method from here but pass in the scanning model, but then we're back to having 4 methods defined that all need to take something in, which seems bad.

Reply:
I believe we have the option of inner classes so if you had an inner class model implementation it could have access to the outer class scanningModel

Reply:
Do you have a link? Because I can't find any examples that actually work to even use them as types...

Reply:
I've not tried this!  But;
```
class classX { 
    private y: number = 0; 

    public getY(): number { return this.y; }

    public utilities = new class {
        constructor(public superThis: classX) {
        }
        public testSetOuterPrivate(target: number) {
            this.superThis.y = target;
        }
    }(this);    
}

const x1: classX = new classX();

alert(x1.getY());

x1.utilities.testSetOuterPrivate(4);
alert(x1.getY());
```

Reply:
You wouldn't need the (this) bit for your case, so you might have something like;
```
public ModelClass = new class {
    constructor(...) {
    }
    ...
}

public makeModel(...) {
    return ModelClass(this);
}
```

Reply:
Scrap that!  If you have to pass in the outer class reference anyway you may as well just do;
```
export class AnotherClass {
  private outer: OBFBoard;
  constructor(outer: OBFBoard) {
    this.outer = outer;
  }
  public doSomething(input: string): string {
    return this.outer.format;
  }
}
```
It doesn't need to be inner

Reply:
Which I guess is back to having to pass around the scanningModel to all of your other model classes anyway.  But is probably the way to do it